### DISCUSSION TITLE
Safely get values from a Map?

### DISCUSSION THREAD

**Discussion by screendriver** [true-myth/true-myth#207] [Ideas]
For Arrays you already have three great functions: [head](https://true-myth.js.org/modules/maybe.html#head-1), [last](https://true-myth.js.org/modules/maybe.html#last-1) and [find](https://true-myth.js.org/modules/maybe.html#find-1). But what about [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)?

```ts
const myMap = new Map<string, string>();
myMap.get('foo'); // string | undefined
```

The only thing that I found was [get](https://true-myth.js.org/modules/maybe.html#get-1) but it seems that `get()` is just working for object literals.

**Comment by chriskrycho** [comment]
Thanks for opening this! In that case, you could use (and for now I would recommend using) `Maybe.of`: `Maybe.of(Map.get('foo'))`. There's a tradeoff in the current package design of adding too many helpers of that variety, because of the cost in package size—and in fact medium-term I think we want to pull the array helpers out into a separate package. I should write up what I've been thinking as a little mini-RFC and get feedback.
